### PR TITLE
fix alerts not retriggered when updating rule

### PR DESCRIPTION
**Pull request by junotx**
**What type of PR is this?**

/kind bug
/area alerting

**What this PR does / why we need it**:

ensure alerts retriggered when updating alerting rule

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubesphere/kubesphere/issues/3422

/cc @benjaminhuo

### CODE REVIEW

File: pkg/models/alerting/rules/ruler.go [hunk 1]
```diff
@@ -102,38 +102,108 @@ func (r *ruleResource) deleteAlertingRules(rules ...*RuleWithGroup) (bool, error
...
 			}
 		}
 	}
 
+	for groupMax++; len(ruleMap) > 0; groupMax++ {
+		g := promresourcesv1.RuleGroup{Name: fmt.Sprintf("%s%d", customRuleGroupDefaultPrefix, groupMax)}
+		var num = customRuleGroupSize
+		if num > 0 {
+			for name, rule := range ruleMap {
+				if num <= 0 {
```
Comment (zryfish) [inline pkg/models/alerting/rules/ruler.go:189]: why there is `num <= 0` since the condition is `num > 0`

Comment (junotx) [reply pkg/models/alerting/rules/ruler.go:189]: `num` is a decreasing var to record the number of rules that can be added to the group

Comment (zryfish) [reply pkg/models/alerting/rules/ruler.go:189]: I see. Isn't it better to make it a function? There are similar codes above.

Comment (junotx) [reply pkg/models/alerting/rules/ruler.go]: It is made now

**Review by benjaminhuo — APPROVED** [review]
/lgtm

**Pull request merged**